app/Jobs/Bank/ProcessBankTransactionsYodlee.php

```diff
@@ -151,14 +157,14 @@ private function processTransactions()
 
         /*Get the user */
         $user_id = $this->company->owner()->id;
-        
+
         /* Unguard the model to perform batch inserts */
         BankTransaction::unguard();
 
         $now = now();
-        
+
         foreach ($transactions as $transaction) {
-            if (BankTransaction::query()->where('transaction_id', $transaction['transaction_id'])->where('company_id', $this->company->id)->withTrashed()->exists()) {
+            if (BankTransaction::query()->where('transaction_id', $transaction['transaction_id'])->where('company_id', $this->company->id)->where('bank_integration_id', $this->bank_integration->id)->withTrashed()->exists()) { // @turbo124 was not scoped to bank_integration_id => from my pov this should be present, because when an account was historized (is_deleted) a transaction can occur multiple (in the archived bank_integration and in the new one
```

Comment:
@turbo124 was not scoped to bank_integration_id => from my pov this should be present, because when an account was historized (is_deleted) a transaction can occur multiple (in the archived bank_integration and in the new one

Reply:
I think this links back to whether we create new integration, or restore existing. This will determine which way to handle this.